dev-docs/rfcs/001_sudo_mode/sudo_mode.md

```diff
@@ -0,0 +1,104 @@
+- We don't currently clear out older sessions but this may be something worth looking into ([internal discussion](https://hackclub.slack.com/archives/C047Y01MHJQ/p1750259883680629)).
+- `UserSession` creation is tracked using the [`public_activity` gem](https://github.com/public-activity/public_activity) so it can be displayed under "Recent activity"
+
+    <img src="login_activity.png" width="200"/>
+    
+    This is configured here https://github.com/hackclub/hcb/blob/cbfccf8975fd8a9c258f916034ef8e7836e80229/app/models/user_session.rb#L47-L48 and displayed here https://github.com/hackclub/hcb/blob/d293e57763729f69a1612606e22095761650eccd/app/controllers/my_controller.rb#L6-L13
+
+## Design considerations
+
+> [!NOTE]
+> We will assume a user is in sudo mode if they are logged in and last authenticated less than 2 hours ago
+
+```mermaid
+flowchart LR
+    A["Log in"] -- Time passes --> B["Initiate sudo action"]
+    B --> C["Authentication older than 2 hours?"]
+    C -- Yes --> D["Re-authenticate"]
+    D --> E["Proceed"]
+    C -- No --> E
+```
+
+**Assumptions**
+
+1. Users who initiate a sudo action and subsequently abandon it (e.g. because they don't have access to their email or TOTP) should remain logged in.
+2. Re-authentication only requires a single factor of the user's choosing
+3. Once they are re-authenticated we won't require them to do so for another two hours
+
+**Open questions**
+
+1. We currently show users a list of their sessions including how long ago they logged in. Should we also surface information about re-authentications?
+
+   <img src="sessions.png" width="200"/>
+2. We send users an email when there is a new login on their account. Would we want to do the same for re-authentications?
+
+   <img src="login_notification_email.png" width="200"/>
+3. How do we want to handle sudo actions for users that are being impersonated? Is re-authenticating the impersonator enough?
+    - _Opinion_: We should strongly consider making this question go away by capping impersonated sessions to 2 hours. I don't think there's a use case for having these persist for longer than that. 
+4. Should re-authentications appear in activities?
+5. How should this work within https://github.com/hackclub/hcb-mobile? Should we return a known error code so the mobile app can prompt the user to re-authenticate?
+
+### How do we model the re-authentication?
+
+**Option 1 - Treat it as a new `Login`**
+
+Given that we have an existing flow for authenticating users, we could treat re-authentication as a new login flow which, once complete, swaps in a new `UserSession`.
+
+We would ideally link the new `Login` to the previous one so we can distinguish it as a re-authentication, apply different rules (e.g. only requiring one factor), and maintain a chain of `Login` records (for auditing purposes).
```

Comment:
I think this a good idea, we could do something like `voided_by_id` to store the ID of the past `Login`